openedx/core/djangoapps/theming/middleware.py

```diff
@@ -0,0 +1,28 @@
+from django.conf import settings
+from django.core.cache import cache
+from django.shortcuts import redirect
+from django.contrib.redirects.models import Redirect
+from openedx.core.djangoapps.theming.helpers import is_request_in_themed_site
+
+
+class ThemeRedirectMiddleware(object):
+    """
+    Redirect users from expired courses' about page to all courses page
+    """
+    def process_request(self, request):
+        """
+        Checks request has been made from theme, redirect the user from expired
+        courses' about page to all courses page by cache or Redirect Model.
+        """
+        host = request.META.get('HTTP_HOST')
+        if host and is_request_in_themed_site():
```

Comment:
@douglashall, I have to check host because there are some testscases (
lms.djangoapps.shoppingcart.tests.test_views.ShoppingCartViewsTests.test_shopping_cart_navigation_link_in_microsite_courseware_page) in which request has no HTTP_HOST.